readme.md

```diff
@@ -313,6 +313,40 @@ array(validation, fields)
 array(fields)
 ```
 
+If you want to use heterogeneous arrays (different types) you can use a `function` instead of a `fields` object:
+
+```js
+array(initialValue => ({
+  _type: required,
+  ...(
+    initialValue._type === 'content' ? { text: required } :
+    initialValue._type === 'image' ? { image: required } :
+    null
+  )
+}))
+```
+
+When rendering the `array` field you can render a different component based on the value of the field:
+
+```jsx
```

Comment:
Moet zeggen dat dit voorbeeld mij meer zegt dan de code die nu in `example/src/domain/Full.js` staat. Misschien is het handig om deze gelijk te trekken? 

Het voorbeeld is meteen een generiek gemaakt component ik weet niet of dat het zo tot de verbeelding spreekt als het voorbeeld hier.